arviz/data/io_pymc3.py

```diff
@@ -34,47 +42,41 @@ def __init__(
             )
             else None
         )
+        if self.observations is not None:
+            self.observations = {obs.name: obs.observations for obs in self.model.observed_RVs}
 
         import pymc3
 
         self.pymc3 = pymc3
 
+    def log_likelihood_vals_point(self, point, var, log_like_fun):
+        """Compute log likelihood for each observed point."""
+        log_like_val = np.atleast_1d(log_like_fun(point))
+        if var.missing_values:
+            log_like_val = log_like_val[~var.observations.mask]
+        return log_like_val
+
     @requires("trace")
+    @requires("model")
     def _extract_log_likelihood(self):
-        """Compute log likelihood of each observation.
-
-        Return None if there is not exactly 1 observed random variable.
-        """
-        # This next line is brittle and may not work forever, but is a secret
-        # way to access the model from the trace.
-        model = self.trace._straces[0].model  # pylint: disable=protected-access
-        if len(model.observed_RVs) != 1:
-            return None, None
-        else:
-            if self.dims is not None:
-                coord_name = self.dims.get(
-                    "log_likelihood", self.dims.get(model.observed_RVs[0].name)
+        """Compute log likelihood of each observation."""
```

Comment:
Where does the final combined logp come from? Usually that should be in the sampler stats already.

Reply:
`log_likelihoods` group contains the `logp_elemwise` of each observed variable stored with the variable name (like `posterior_predictive` and `observed_data` groups that have matching variables. And in addition, it stores the `model_logp` stored as `lp`.